### PR TITLE
More spec for ENV[]

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: core/env/element_reference_spec.rb [hunk 1]
```diff
@@ -19,6 +19,17 @@
     ENV[@variable].frozen?.should == true
   end
 
+  it "coerces a non-string name with #to_str" do
+    ENV[@variable] = "bar"
```
Comment (eregon) [inline core/env/element_reference_spec.rb]: I think this line should be removed.

Comment (BurdetteLamar) [reply core/env/element_reference_spec.rb]: Not clear to me which line, nor why.

Comment (eregon) [reply core/env/element_reference_spec.rb]: This line above `ENV[@variable] = "bar"` already sets the environment variable, and so the mock below is not actually tested for whether it sets the value correctly. Just removing this line fixes that.

Comment (BurdetteLamar) [reply core/env/element_reference_spec.rb]: I'm very confused.  Isn't ```ENV#[]``` a getter?  So the coercion is applied to ```k```, producing the value of ```@variable``` as the argument, which returns ```"bar"```.

If I remove ```ENV[@variable] = "bar"``` I get an error:  ```Expected nil == "bar" to be truthy but was false```.

Or is that not the line to remove?

Comment (eregon) [reply core/env/element_reference_spec.rb]: You're completely right, I'm sorry, I'm tired.
I thought I saw some `ENV[k] = ...` below but obviously there isn't.

**Pull request merged**